ipv6 support

if the record is moved client cant parse ipv6 address:

```
scredis.exceptions.RedisProtocolException: Could not decode response
	at scredis.io.DecoderActor$$anonfun$receive$1.applyOrElse(DecoderActor.scala:36)
	at akka.actor.Actor.aroundReceive(Actor.scala:533)
	at akka.actor.Actor.aroundReceive$(Actor.scala:531)
	at scredis.io.DecoderActor.aroundReceive(DecoderActor.scala:12)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:573)
	at akka.actor.ActorCell.invoke(ActorCell.scala:543)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:269)
	at akka.dispatch.Mailbox.run(Mailbox.scala:230)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:242)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: scredis.exceptions.RedisProtocolException: invalid format for MOVED or ASK error: MOVED 7310 xxxx:xxxx:xxx:xxxx:xxxx:x:x:xxx:6379
	at scredis.protocol.Protocol$.decodeError(Protocol.scala:296)
	at scredis.protocol.Protocol$.decode(Protocol.scala:276)
	at scredis.io.DecoderActor$$anonfun$receive$1.applyOrElse(DecoderActor.scala:31)
	... 12 common frames omitted
Caused by: java.lang.NumberFormatException: For input string: "xxxx:xxx:xxxx:xxxx:x:x:xxx:6379"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:580)
	at java.lang.Integer.parseInt(Integer.java:615)
	at scredis.protocol.Protocol$.decodeMoveAsk(Protocol.scala:315)
	at scredis.protocol.Protocol$.decodeError(Protocol.scala:293)
	... 14 common frames omitted
```

client version: 2.3.4

--- a/src/main/scala/scredis/protocol/Protocol.scala
+++ b/src/main/scala/scredis/protocol/Protocol.scala
@@ -307,10 +307,11 @@ object Protocol extends LazyLogging {
 
   private[scredis] def decodeMoveAsk(err: String, msg: String): ClusterError = {
     // shape of response: -MOVED 3999 127.0.0.1:6381
+    // when IP v6 used: -MOVED 3999 2001:db8::1:0:0:1:6381
     // let's just omit error handling here and leave it to the caller
     val space = msg.indexOf(' ')
     val hashSlot = Integer.parseInt(msg.substring(0, space))
-    val colon = msg.indexOf(':',space)
+    val colon = msg.lastIndexOf(':')
     val host = msg.substring(space + 1, colon)
     val port = Integer.parseInt(msg.substring(colon+1))
 

--- a/src/test/scala/scredis/protocol/ProtocolSpec.scala
+++ b/src/test/scala/scredis/protocol/ProtocolSpec.scala
@@ -143,6 +143,20 @@ class ProtocolSpec extends AnyWordSpec with ScalaCheckDrivenPropertyChecks with 
         }
       }
     }
+
+    "decode MOVED error IP v6" in {
+      val slot = 3999
+      val host = "2001:db8::1:0:0:1"
+      val port = 6381
+      val errMsg = s"MOVED $slot $host:$port"
+      val err = Protocol.decodeError(errMsg)
+      inside (err) { case (ClusterErrorResponse(Moved(s, h, p), msg)) =>
+        s should be (slot)
+        h should be (host)
+        p should be (port)
+        msg should include ("MOVED")
+      }
+    }
   }
 
 }

